utils/core-modules/contracts/interfaces/IAssociatedSystemsModule.sol

```diff
@@ -1,9 +1,42 @@
 //SPDX-License-Identifier: MIT
 pragma solidity ^0.8.0;
 
-/// @title Allows for the registration and tracking of auxillery contracts which also follow the proxy architecture
+/**
+ * @title Module for connecting a system with other associated systems.
+ *
+ * Associated systems can me managed or unmanaged.
+ *
+ * Managed systems are connected via a proxy, which means that their implementation can be updated, and the system controls the execution context of the associated system. Example, an snxUSD token connected to the system, and controlled by the system.
+ *
+ * Unmanaged systems are just addresses tracked by the system, for which it has no control whatsoever. Example, Uniswap v3, Curve, etc.
+ *
+ * Associated systems, managed or unmanaged, become available to all system modules for communication and interaction, but as opposed to inter-modular communications, associated system interactions will require the use of `CALL`.
+ */
 interface IAssociatedSystemsModule {
-    /// @notice create or initialize a new token
+    /**
+     * @notice Emitted when an associated system is set.
+     * @param kind The type of associated system (managed ERC20, managed ERC721, unmanaged, etc - See the AssociatedSystem util).
+     * @param id The bytes32 identifier of the associated system.
+     * @param proxy The main external contract address of the associated system.
+     * @param impl The address of the implementation of the associated system (if not behind a proxy, will equal `proxy`).
+     */
+    event AssociatedSystemSet(
+        bytes32 indexed kind,
+        bytes32 indexed id,
+        address proxy,
+        address impl
+    );
+
+    /**
+     * @notice Creates or initializes a managed associated ERC20 token.
+     * @param id The bytes32 identifier of the associated system.
+     * @param name The token name that will be used to initialize the proxy.
+     * @param symbol The token symbol that will be used to initialize the proxy.
+     * @param decimals The token decimals that will be used to initialize the proxy.
+     * @param impl The first/new ERC20 implementation of the proxy.
+     *
+     * Note: If the id is new, it will create a new proxy and set its first implementation. Otherwise, it will upgrade the implementation of the existing proxy.
```

Comment:
This is a little confusing because above you say the "first/new" implementation, but here it reads like it could be an existing one?

Reply:
Also, these always assume 18 decimals I think? Probably worth mentioning somewhere

Reply:
> This is a little confusing because above you say the "first/new" implementation, but here it reads like it could be an existing one?
Made a change, please review.

> Also, these always assume 18 decimals I think? Probably worth mentioning somewhere
This module does not make that assumption, but the system does. Which means I don't think this clarification needs to be made here.